### PR TITLE
Feature/create target

### PR DESCRIPTION
**Pull request by JorgeMichelena**

### CODE REVIEW

File: target/settings.py [hunk 1]
```diff
@@ -101,9 +103,16 @@
...
+        'ENGINE' : 'django.contrib.gis.db.backends.postgis',
+        'NAME' : 'target',
+        'USER' : 'jorgemichelena',
+        'PASSWORD' : '',
+        'HOST' : '',
+        'PORT' : '',
+        'TEST' : {
+            'NAME' : 'testing_database',
+        },
+    },
```
Comment (kaozdl) [inline target/settings.py]: you shouldnt commit connection info, try to use `os.getenv()` for this kind of stuff

File: target/settings.py [hunk 2]
```diff
@@ -163,6 +172,5 @@
 
 
 # File management settings
-
 MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
 MEDIA_URL = 'api/v1/media/'
```
Comment (kaozdl) [inline target/settings.py]: missing newline

File: target/settings.py [hunk 3]
```diff
@@ -20,7 +20,7 @@
 # See https://docs.djangoproject.com/en/3.0/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = '!f7k)2j8c!@o!ljx7deoaylhvv)1b$n(np-9l-x7_!4h#*ux*8'
+SECRET_KEY = os.getenv('SECRET_KEY')
 
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
```
Comment (jiss2891) [inline target/settings.py]: This could be grabbed using os.getenv too.

File: targets/models.py [hunk 4]
```diff
@@ -1,5 +1,17 @@
...
+    picture = models.ImageField(upload_to='media/topic_pictures/', null=True)
+    def __str__(self):
+        return self.name
+
+class Target(gis_models.Model):
+    user = models.ForeignKey('users.User', on_delete=models.CASCADE, related_name='targets')
+    topic = models.ForeignKey('Topic', on_delete=models.CASCADE)
+    title = models.CharField(max_length=20)
+    radius = models.PositiveIntegerField()
+    location = gis_models.PointField(geography=True, validators=[validate_coordinates])
```
Comment (jiss2891) [inline targets/models.py]: missing newline here.

File: targets/validators.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    latitude = value.x
+    longitude = value.y
+    if latitude>90 or latitude<-90:
+        raise ValidationError(
+            _('Latitude must be a real number between -90 and 90'), 
+        )
+    if longitude>180 or longitude<-180:
+        raise ValidationError(
+            _('Longitude must be a real number between -180 and 180'), 
+        )
```
Comment (jiss2891) [inline targets/validators.py]: Same here.

File: targets/validators.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
+
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+def validate_coordinates(value):
+    latitude = value.x
+    longitude = value.y
+    if latitude>180 or latitude<-180:
+        raise ValidationError(
+            _('Latitude must be a real number between -180 and 180'),
```
Comment (GuilleLeopold) [inline targets/validators.py]: Do we have internationalization? @JorgeMichelena @kaozdl @jiss2891

File: targets/validators.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+def validate_coordinates(value):
+    latitude = value.x
+    longitude = value.y
+    if latitude>180 or latitude<-180:
+        raise ValidationError(
+            _('Latitude must be a real number between -180 and 180'), 
+        )
+    if longitude>90 or longitude<-90:
+        raise ValidationError(
+            _('Longitude must be a real number between -90 and 90'),
```
Comment (GuilleLeopold) [inline targets/validators.py]: same here

File: targets/models.py [hunk 8]
```diff
@@ -1,5 +1,20 @@
 from django.db import models
+from django.contrib.gis.db import models as gis_models
+from targets.validators import validate_coordinates
+
 
 class Topic(models.Model):
     name = models.CharField(max_length=30, primary_key=True)
-    picture = models.TextField(default='')
\ No newline at end of file
+    picture = models.TextField(default='')
```
Comment (kaozdl) [inline targets/models.py]: why textField and not ImageField?

File: targets/validators.py (hunk 6 above)
Comment (kaozdl) [reply targets/validators.py]: he's using it with the `_(...)`

File: targets/models.py (hunk 8 above)
Comment (JorgeMichelena) [reply targets/models.py]: I am going to convert images to base64 and save them as strings.

**Pull request merged**